### PR TITLE
DOCSP-24872 adds FLE options

### PR DESCRIPTION
**Pull request by jmd-mongo**
# STAGING

[CSFLE Options](https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-shell/docsworker-xlarge/DOCSP-24872/reference/options/#client-side-field-level-encryption-options)

# JIRA

https://jira.mongodb.org/browse/DOCSP-24872

### CODE REVIEW

File: source/reference/options.txt [hunk 1]
```diff
@@ -477,6 +477,87 @@ Session Options
...
+.. option:: --keyVaultNamespace <string>
+
+   The full namespace (``<database>.<collection>``) of the collection 
+   used as a key vault for :ref:`manual-csfle-feature`. 
+   :option:`--keyVaultNamespace` is required for enabling client-side 
+   field level encryption. for the :program:`mongosh` shell session. 
+   :program:`mongosh` creates the specified namespace if it does not
+   exist.
+
+   :option:`--keyVaultNamespace` requires *all* of the following command
```
Comment (davemungo) [inline source/reference/options.txt:555]: As above

File: source/reference/options.txt [hunk 2]
```diff
@@ -477,6 +477,87 @@ Session Options
...
+   specifying an environmental variable to :option:`--awsAccessKeyId`. 
+
+.. option:: --awsSecretAccessKey <string>
+
+   An AWS `Secret Key <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html>`__
+   associated to the specified :option:`--awsAccessKeyId`.
+
+   :option:`--awsSecretAccessKey` is required for enabling 
+   :ref:`manual-csfle-feature` for the :program:`mongosh` session. 
+   :option:`--awsSecretAccessKey` requires *all* of the following
```
Comment (davemungo) [inline source/reference/options.txt:510]: As above

File: source/reference/options.txt [hunk 3]
```diff
@@ -477,6 +477,87 @@ Session Options
...
+
+   To mitigate the risk of leaking access keys into logs, consider 
+   specifying an environmental variable to :option:`--awsAccessKeyId`. 
+
+.. option:: --awsSecretAccessKey <string>
+
+   An AWS `Secret Key <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html>`__
+   associated to the specified :option:`--awsAccessKeyId`.
+
+   :option:`--awsSecretAccessKey` is required for enabling
```
Comment (davemungo) [inline source/reference/options.txt:508]: Consider:
  is required to enable

File: source/reference/options.txt [hunk 4]
```diff
@@ -477,6 +477,87 @@ Session Options
...
+   specifying an environmental variable to 
+   :option:`--awsSecretAccessKey`. 
+
+.. option:: --awsSessionToken <string>
+
+   An AWS `Session Token 
+   <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html>`__
+   associated to the specified :option:`--awsAccessKeyId`.
+
+   :option:`--awsSessionToken` is required for enabling
```
Comment (davemungo) [inline source/reference/options.txt:530]: As above

**Review by davemungo — APPROVED** [review]
@jmd-mongo thanks for the updates this LGTM

This page appears to reuse the existing text from the mongo shell page. No problem there. It might make sense to confirm with the tech reviewer that there are no differences between the mongo and mongosh shells. 

There are  one or two small suggestions  to clean up the existing text. No problem to ignore them if you don't want to get caught up in the older material.

File: source/reference/options.txt [hunk 5]
```diff
@@ -477,6 +477,87 @@ Session Options
 Client-Side Field Level Encryption Options
 ------------------------------------------
 
-For information on Client-Side Field Level Encryption Options,
-refer to the :manual:`MongoDB Manual
-</reference/program/mongo/#client-side-field-level-encryption-options>`.
+.. option:: --awsAccessKeyId <string>
+
+   An AWS `Access Key
```
[the remark this answers is not in the document]
Comment (jmd-mongo) [reply source/reference/options.txt:482]: Done.

File: source/reference/options.txt [hunk 6]
```diff
@@ -477,6 +477,87 @@ Session Options
...
-</reference/program/mongo/#client-side-field-level-encryption-options>`.
+.. option:: --awsAccessKeyId <string>
+
+   An AWS `Access Key 
+   <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html>`__
+   associated with an IAM user who has ``List`` and ``Read`` permissions 
+   for the AWS Key Management Service (KMS). :program:`mongosh` uses the 
+   specified :option:`--awsAccessKeyId` to access the KMS. 
+
+   :option:`--awsAccessKeyId` is required for enabling
```
[the remark this answers is not in the document]
Comment (jmd-mongo) [reply source/reference/options.txt:488]: Done.

File: source/reference/options.txt [hunk 7]
```diff
@@ -477,6 +477,87 @@ Session Options
...
+
+   An AWS `Access Key 
+   <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html>`__
+   associated with an IAM user who has ``List`` and ``Read`` permissions 
+   for the AWS Key Management Service (KMS). :program:`mongosh` uses the 
+   specified :option:`--awsAccessKeyId` to access the KMS. 
+
+   :option:`--awsAccessKeyId` is required for enabling 
+   :ref:`manual-csfle-feature` for the :program:`mongosh` shell session. 
+   :option:`--awsAccessKeyId` requires *all* of the following command
```
[the remark this answers is not in the document]
Comment (jmd-mongo) [reply source/reference/options.txt:490]: Updated.

File: source/reference/options.txt (hunk 1 above)
Comment (jmd-mongo) [reply source/reference/options.txt:555]: Done.

File: source/reference/options.txt (hunk 2 above)
Comment (jmd-mongo) [reply source/reference/options.txt:510]: Done.

File: source/reference/options.txt (hunk 3 above)
Comment (jmd-mongo) [reply source/reference/options.txt:508]: Done.

File: source/reference/options.txt (hunk 4 above)
Comment (jmd-mongo) [reply source/reference/options.txt:530]: Done.

File: source/reference/options.txt [hunk 8]
```diff
@@ -477,6 +477,87 @@ Session Options
...
+   If :option:`--awsAccessKeyId` is omitted, use the :method:`Mongo()` 
+   constructor within the shell session to enable client-side field 
+   level encryption.
+
+   To mitigate the risk of leaking access keys into logs, consider 
+   specifying an environmental variable to :option:`--awsAccessKeyId`. 
+
+.. option:: --awsSecretAccessKey <string>
+
+   An AWS `Secret Key <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html>`__
```
Comment (Anemy) [inline source/reference/options.txt:505]: This is a small thing I noticed, looks like we don't include `AWS` in the link here, while we do include `AWS` in the link in the description for `--awsAccessKeyId`. Should we go for one option to keep it consistent? Also fine as is.

**Review by Anemy — APPROVED** [review]

**Pull request merged**